app/cisc/lib.go

```diff
@@ -0,0 +1,172 @@
+	}
+	return nil
+}
+
+// Saves the clientApp in the configfile - refuses to save an empty file.
+func saveConfig() error {
+	if clientApp == nil {
+		return errors.New("Cannot save empty clientApp")
+	}
+	buf, err := network.MarshalRegisteredType(clientApp)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(configFile, buf, 0660)
+}
+
+// Asserts that the clientApp exists, else fatals
+func assertCA() {
+	if clientApp == nil {
+		log.Fatal("Couldn't load config-file", configFile, "or it was empty.")
+	}
+}
+
+// Reads the group-file and returns it
+func getGroup(c *cli.Context) *config.Group {
+	gfile := c.Args().Get(0)
+	gr, err := os.Open(gfile)
+	log.ErrFatal(err)
+	groups, err := config.ReadGroupDescToml(gr)
+	gr.Close()
+	if groups == nil || groups.Roster == nil || len(groups.Roster.List) == 0 {
+		log.Fatal("No servers found in roster from", gfile)
+	}
+	return groups
+}
+
+// kvGetKeys returns the keys up to the next ":". If given a slice of keys, it
+// will return sub-keys.
+func kvGetKeys(keys ...string) []string {
+	var ret []string
+	start := strings.Join(keys, ":")
+	if len(start) > 0 {
+		start += ":"
+	}
+	for k := range clientApp.Config.Data {
+		if strings.HasPrefix(k, start) {
+			// Create subkey
+			subkey := strings.TrimPrefix(k, start)
+			subkey = strings.SplitN(subkey, ":", 2)[0]
+			ret = append(ret, subkey)
+		}
+	}
+	return sortUniq(ret)
+}
+
+// kvGetValue returns the value of the key
+func kvGetValue(keys ...string) string {
+	key := strings.Join(keys, ":")
+	for k, v := range clientApp.Config.Data {
+		if k == key {
+			return v
+		}
+	}
+	return ""
+}
+
+// kvGetIntKeys returns the keys in the middle of prefix and suffix
+func kvGetIntKeys(prefix, suffix string) []string {
+	var ret []string
+	if len(prefix) > 0 {
+		prefix += ":"
+	}
+	if len(suffix) > 0 {
+		suffix = ":" + suffix
+	}
+	for k := range clientApp.Config.Data {
+		if strings.HasPrefix(k, prefix) && strings.HasSuffix(k, suffix) {
+			interm := strings.TrimPrefix(k, prefix)
+			interm = strings.TrimSuffix(interm, suffix)
+			if !strings.Contains(interm, ":") {
+				ret = append(ret, interm)
+			}
+		}
+	}
+	return sortUniq(ret)
+}
+
+// sortUniq sorts the slice of strings and deletes duplicates
+func sortUniq(slice []string) []string {
+	sorted := make([]string, len(slice))
+	copy(sorted, slice)
+	sort.Strings(sorted)
+	var ret []string
+	for i, s := range sorted {
+		if i == 0 || s != sorted[i-1] {
+			ret = append(ret, s)
+		}
+	}
+	return ret
+}
+
+// MakeSSHKeyPair make a pair of public and private keys for SSH access.
+// Public key is encoded in the format for inclusion in an OpenSSH authorized_keys file.
+// Private Key generated is PEM encoded
+// StackOverflow: Greg http://stackoverflow.com/users/328645/greg in
+// http://stackoverflow.com/questions/21151714/go-generate-an-ssh-public-key
+// No licence added
+func makeSSHKeyPair(pubKeyPath, privateKeyPath string) error {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 1024)
```

Comment:
I'll try if 2048 work, too. I copy/pasted the code from Stackoverflow, so I don't know if it will work...